redisinsight/ui/src/pages/vector-search/query/useQuery.ts

```diff
@@ -1,184 +1,40 @@
 import { useCallback, useEffect, useRef, useState } from 'react'
 import { useParams } from 'react-router-dom'
-import { chunk, reverse } from 'lodash'
-import {
-  Nullable,
-  getCommandsForExecution,
-  getExecuteParams,
-  isGroupResults,
-  isSilentMode,
-} from 'uiSrc/utils'
+import { useDispatch, useSelector } from 'react-redux'
 import { CodeButtonParams } from 'uiSrc/constants'
 import {
   RunQueryMode,
   ResultsMode,
-  CommandExecutionUI,
-  CommandExecution,
+  CommandExecutionType,
 } from 'uiSrc/slices/interfaces'
-import { PIPELINE_COUNT_DEFAULT } from 'uiSrc/constants/api'
-import {
-  addCommands,
-  clearCommands,
-  findCommand,
-  removeCommand,
-} from 'uiSrc/services/workbenchStorage'
 import {
-  createErrorResult,
-  createGroupItem,
-  executeApiCall,
-  generateCommandId,
-  limitHistoryLength,
-  loadHistoryData,
-  prepareNewItems,
-  scrollToElement,
-  sortCommandsByDate,
-} from './utils'
+  fetchWBHistoryAction,
+  workbenchResultsSelector,
+  sendWbQueryAction,
+  deleteWBCommandAction,
+  clearWbResultsAction,
+  fetchWBCommandAction,
+  toggleOpenWBResult,
+} from 'uiSrc/slices/workbench/wb-results'
+import { Nullable } from 'uiSrc/utils'
 
 const useQuery = () => {
   const { instanceId } = useParams<{ instanceId: string }>()
   const scrollDivRef = useRef<HTMLDivElement>(null)
+  const dispatch = useDispatch()
 
   const [query, setQuery] = useState('')
-  const [items, setItems] = useState<CommandExecutionUI[]>([])
-  const [clearing, setClearing] = useState(false)
-  const [processing, setProcessing] = useState(false)
-  const [isLoaded, setIsLoaded] = useState(false)
+  const { items, clearing, processing, isLoaded } = useSelector(
+    workbenchResultsSelector,
+  )
 
   const resultsMode = ResultsMode.Default
   const activeRunQueryMode = RunQueryMode.ASCII
+  const executionType = CommandExecutionType.Search
```

Comment:
Basically, we need to rely on the current implementation for the history manipulation (and automatically switch between SQLite and IndexDB based on the feature flag), so we should go for the existing helpers instead of building them again.